docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,769 @@
+code_ to authenticate the device. You can find these values in the logging
+terminal of the device (for instance, UART). For example:
+
+```
+I: 254 [DL]Device Configuration:
+I: 257 [DL] Serial Number: TEST_SN
+I: 260 [DL] Vendor Id: 65521 (0xFFF1)
+I: 263 [DL] Product Id: 32768 (0x8000)
+I: 267 [DL] Hardware Version: 1
+I: 270 [DL] Setup Pin Code: 20202021
+I: 273 [DL] Setup Discriminator: 3840 (0xF00)
+I: 278 [DL] Manufacturing Date: (not set)
+I: 281 [DL] Device Type: 65535 (0xFFFF)
+```
+
+In above printout, the _discriminator_ is 3840 (0xF00) and the _setup PIN code_
+is equal to 20202021 accordingly.
+
+### Step 5: Commission Matter accessory device into existing network
+
+#### Commissioning into Thread network over Bluetooth LE
+
+To commission the device to the existing Thread network run the following
+command:
+
+```
+./BUILD_PATH/chip-tool pairing ble-thread <node_id> hex:<operational_dataset> <pin_code> <discriminator>
+```
+
+where:
+
+-   _<node_id\>_ is the user-defined ID of the node being commissioned,
+-   _<operational_dataset\>_ is the Operational Dataset determined in step 3,
+-   _<pin_code\>_ and _<discriminator\>_ are device specific keys determined in
+    step 4
+
+#### Commissioning into Wi-Fi network over Bluetooth LE
+
+```
+./BUILD_PATH/chip-tool pairing ble-wifi <node_id> <ssid> <password> <pin_code> <discriminator>
+
+```
+
+where:
+
+-   _<node_id\>_ is the user-defined ID of the node being commissioned,
+-   _<ssid\>_ and _<password\>_ are credentials determined in step 3,
+-   _<pin_code\>_ and _<discriminator\>_ are device specific keys determined in
+    step 4
+
+If the hexadecimal format is preferred the `hex:` prefix shall be used, i.e:
+
+```
+./BUILD_PATH/chip-tool pairing ble-wifi <node_id> hex:<ssid> hex:<password> <pin_code> <discriminator>
+
+```
+
+The _<node_id>_ can be simply provided as a hexadecimal value with `0x` prefix.
+
+After connecting the device over Bluetooth LE, the controller will go through
+the following stages:
+
+-   Establishing a secure connection that completes the PASE
+    (Password-Authenticated Session Establishment) session using SPAKE2+
+    protocol and results in printing the following log:
+
+        ```
+        Secure Session to Device Established
+        ```
+
+-   Providing the device with a network interface using ZCL Network
+    Commissioning cluster commands, and the network pairing credentials from
+    step 3
+-   Discovering the IPv6 address of the Matter accessory using the SRP (Service
+    Registration Protocol) for Thread devices, or the mDNS (Multicast Domain
+    Name System) protocol for Wi-Fi or Ethernet devices. It results in printing
+    log that indicates that the node address has been updated. The IPv6 address
+    of the device is cached in the controller for later usage.
+-   Closing the Bluetooth LE connection, as the commissioning process is
+    finished and the `chip-tool` is now using only the IPv6 traffic to reach the
+    device.
```

Comment:
This also belongs to a description of how the controller works in general. Finding it in steps is like stopping in the middle of an action sequence in a Christopher Nolan movie to discuss some science stuff. I think the user does this step and only sees the log from line 251, no?

Reply:
This is a copy-paste from [Python Chip Controller guide](https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/python_chip_controller_building.md) I personally don't have a strong opinion on if we should add some theory background in such guides. On the other hand maybe we should have a document describing briefly this kind of stuff, i.e. After discussion with @ArekBalysNordic we agreed to move the binding theory from light-switch-app docs here (Adding a binding table to the `binding` cluster).

Reply:
For now I will remove the science stuff from here :)